### PR TITLE
feat(search): use elastic search nlp vector search

### LINKED ISSUE #1317: Utiliser les nouvelles fonction Elasticsearch pour faire de la recherche semantic
travail en cours sur la PR #1309

### PR DESCRIPTION
**Pull request by lionelB**
~~wip~~
use elastic new script_query to search using nlp vector
to do : 
- [ ] ~~add a new nlp_api endpoint for populate document index with vector~~
- [ ] ~~remove dump from nlp_api~~~
- [x] use ci to build  data dump
- [x] add a endpoint in nlp api to transform a query into vector

fix #1317 and #1318

### CODE REVIEW

File: packages/code-du-travail-api/src/server/routes/search/search.sem.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  return {
+    size: size,
+    _source: ["title", "source", "slug", "anchor", "url"],
+    query: {
+      script_score: {
+        query: {
+          bool: {
+            filter: {
+              bool: {
+                should: sources.map(source => ({ term: { source } }))
```
Comment (revolunet) [inline packages/code-du-travail-api/src/server/routes/search/search.sem.js:14]: what if sources is empty ?

File: packages/code-du-travail-data/indexing/index.js [hunk 2]
```diff
@@ -109,11 +110,50 @@ async function main() {
...
+      THEMES_INDEX_NAME,
+      CDTN_CCN_NAME,
+      ANNUAIRE_INDEX_NAME
+    ];
+    return patterns.some(pattern => index.startsWith(`${pattern}-`));
+  }
+
+  const { body: indices } = await client.cat.indices({ format: "json" });
+
+  const pIndicesToDelete = indices
```
Comment (revolunet) [inline packages/code-du-travail-data/indexing/index.js]: this critical function should have some TUs

File: packages/code-du-travail-nlp/api/sem_search.py [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
 import numpy as np
 import tf_sentencepiece
 import unicodedata
-
 from typing import List
+import time
 
-content_path = "./data/content.json"
-stops_path = "./data/stops.txt"
+stops_path = "../data/stops.txt"
```
Comment (revolunet) [inline packages/code-du-travail-nlp/api/sem_search.py:12]: est-ce que ce sont les memes stops que https://github.com/SocialGouv/code-du-travail-numerique/tree/master/packages/code-du-travail-data/dataset/stop_words ?

File: packages/code-du-travail-api/src/server/routes/search/search.sem.js (hunk 1 above)
Comment (lionelB) [reply packages/code-du-travail-api/src/server/routes/search/search.sem.js:14]: don't really know :) I will test that

File: packages/code-du-travail-data/indexing/index.js (hunk 2 above)
Comment (lionelB) [reply packages/code-du-travail-data/indexing/index.js]: ahah true ! I wonder if I should move that to a dedicated function

File: packages/code-du-travail-nlp/api/sem_search.py (hunk 3 above)
Comment (lionelB) [reply packages/code-du-travail-nlp/api/sem_search.py:12]: je crois pas. À va voir avec Armand si on peut les merger

File: packages/code-du-travail-api/src/server/routes/search/search.sem.js (hunk 1 above)
Comment (lionelB) [reply packages/code-du-travail-api/src/server/routes/search/search.sem.js:14]: source should not be empty, I will throw an error

File: packages/code-du-travail-nlp/api/sem_search.py (hunk 3 above)
Comment (lionelB) [reply packages/code-du-travail-nlp/api/sem_search.py:12]: je pense qu'on fera un PR a part pour traiter cette question

File: packages/code-du-travail-data/indexing/index.js (hunk 2 above)
Comment (revolunet) [reply packages/code-du-travail-data/indexing/index.js]: `const deleteOldIndices = (client, currentIndices) => {}` ?

File: packages/code-du-travail-nlp/api/sem_search.py (hunk 3 above)
Comment (revolunet) [reply packages/code-du-travail-nlp/api/sem_search.py:12]: dac. une petite issue en memo ?

File: packages/code-du-travail-data/indexing/index.js (hunk 2 above)
Comment (lionelB) [reply packages/code-du-travail-data/indexing/index.js]: sound good !

**Review by revolunet — APPROVED** [review]

**Pull request merged**